### ISSUE TITLE
foreground notifications not working if used with react-navigation

### ISSUE THREAD

**Issue by sarangzambare** [wix/react-native-notifications#847]
If I use react-navigation in my app:
- Background notifications work normally
- Foreground notifications don't work - app hangs - although the console log associated with `Notifications.events().registerNotificationReceivedForeground` does get activated

If I do not use react-navigation:
- everything works well

In the docs, it says to refer to the wiki for proper integration with react-native-navigation, but i don't see any relevant wiki there. 

Please help.

Platform: ios 15.3.1
react-native-notifications@4.2.4

**Comment by sarangzambare** [comment]
The issue was because of accidently registering onRecieve call back twice because of poor redux state management. I corrected it and its working now. However, the function associated with background notfif `(Notifications.events().registerNotificationReceivedBackground)`

is still not getting activated when the app is minimized. For that I will create a separate issue after some debugging.

For anyone facing issues: please check that you are registering your callbacks only once and they are not getting registered again because of other factors like redux/ useState etc . thanks !
